### PR TITLE
dummyfs/ext2/jffs2: add mtStat message handling

### PR DESCRIPTION
**Pull request by Maxez**
## Description

Added `mtStat` message handling to filesystems.

## Motivation and Context

[JIRA: DTR-100](https://jira.phoenix-rtos.com/browse/DTR-100)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: ia32-generic, armv7a7-imx6ull.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: dummyfs/srv.c [hunk 1]
```diff
@@ -335,6 +335,10 @@ int main(int argc, char **argv)
 				msg.o.io.err = dummyfs_readdir(ctx, &msg.i.readdir.dir, msg.i.readdir.offs,
 					msg.o.data, msg.o.size);
 				break;
+
+			case mtStat:
+				msg.o.io.err = dummyfs_statfs(ctx, msg.o.data);
```
Comment (agkaminski) [inline dummyfs/srv.c:340]: We should check if msg.o.data != NULL and if msg.o.size == sizeof(struct statvfs)

File: ext2/libext2.c [hunk 2]
```diff
@@ -173,6 +173,10 @@ int libext2_handler(void *fdata, msg_t *msg)
 	case mtUnlink:
 		msg->o.io.err = ext2_unlink(fs, msg->i.ln.dir.id, msg->i.data, (uint8_t)strlen(msg->i.data));
 		break;
+
+	case mtStat:
+		msg->o.io.err = ext2_statfs(fs, msg->o.data);
```
Comment (agkaminski) [inline ext2/libext2.c:178]: As with dummyfs, arg check missing

File: jffs2/jffs2.c [hunk 3]
```diff
@@ -947,6 +975,10 @@ int jffs2lib_message_handler(void *partition, msg_t *msg)
 				msg->o.data, msg->o.size);
 		break;
 
+	case mtStat:
+		msg->o.io.err = jffs2_srv_statfs(p, msg->o.data);
```
Comment (agkaminski) [inline jffs2/jffs2.c:979]: As with dummyfs, arg check missing

File: dummyfs/srv.c (hunk 1 above)
Comment (Maxez) [reply dummyfs/srv.c]: Done.

File: ext2/libext2.c (hunk 2 above)
Comment (Maxez) [reply ext2/libext2.c]: Done.

File: jffs2/jffs2.c (hunk 3 above)
Comment (Maxez) [reply jffs2/jffs2.c]: Done.

**Review by agkaminski — APPROVED** [review]

**Pull request merged**